### PR TITLE
MODFISTO-48 Adjustments to allow listing budgets by group/fiscalYear/ledger

### PR DESCRIPTION
**Pull request by pautau-s**
## Purpose
[MODFISTO-48](https://issues.folio.org/browse/MODFISTO-48) Adjustments to allow listing budgets by group/fiscalYear/ledger

## Approach
- created group_budgets_view
- added ability to search budgets by groups fields
- added  foreign key `group_fund_fiscal_years.budgetId->budget.id` to `group_fund_fiscal_years` table

#### TODOS and Open Questions
- [ ] Point to merge commit once folio-org/acq-models#190 is merged.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/resources/templates/db_scripts/schema.json [hunk 1]
```diff
@@ -98,7 +105,7 @@
     },
     {
       "tableName": "fiscal_year",
-      "fromModuleVersion": 1.0,
+      "fromModuleVersion": "mod-finance-storage-4.0.0",
```
Comment (KVupp) [inline src/main/resources/templates/db_scripts/schema.json]: moving from 1.0 to 4.0?

Comment (KVupp) [reply src/main/resources/templates/db_scripts/schema.json]: was anything in these tables updated?

**Comment by craigmcnally** [comment]
What about the ModuleDescriptor?

File: src/test/java/org/folio/rest/impl/GroupBudgetTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    GroupFundFiscalYear groupFundFiscalYear = new GroupFundFiscalYear()
+      .withFiscalYearId(fiscalYear.getId())
+      .withFundId(budget.getFundId())
+      .withBudgetId(budget.getId())
+      .withGroupId(responseGroup.getId());
+
+    BudgetCollection group1BudgetCollection = getData(getEndpoint(FinanceStorageGroupBudgets.class) + EMPTY_ACQ_UNITS_QUERY, VIEW_SEARCH_TENANT_HEADER).as(BudgetCollection.class);
+    BudgetCollection group2BudgetCollection = getData(getEndpoint(FinanceStorageGroupBudgets.class) + String.format(QUERY_ACQ_UNIT_BY_ID, acqUnitId), VIEW_SEARCH_TENANT_HEADER).as(BudgetCollection.class);
+    assertThat(group1BudgetCollection.getBudgets(), hasSize(GROUP_FUND_FY.getInitialQuantity()));
+    assertThat(group2BudgetCollection.getBudgets(), hasSize(0));
```
Comment (KVupp) [inline src/test/java/org/folio/rest/impl/GroupBudgetTest.java]: shouldn't this have a size of 1 as you are adding a group with acqUnitId in line 63?

Can you also please add few comments along with these assertions so it's easy to follow? Thanks

File: src/main/resources/templates/db_scripts/schema.json (hunk 1 above)
Comment (KVupp) [reply src/main/resources/templates/db_scripts/schema.json]: our pom is still at 3.1.0-SNAPSHOT? will these changes be applied in-spite of that?
Also, I thought we wanted to increase the module version with any changes immediately and not during the release? aren't we following that?

File: src/test/java/org/folio/rest/impl/GroupBudgetTest.java (hunk 2 above)
Comment (pautau-s) [reply src/test/java/org/folio/rest/impl/GroupBudgetTest.java]: Done

File: src/main/resources/templates/db_scripts/group_budget_view.sql [hunk 3]
```diff
@@ -0,0 +1,4 @@
+CREATE OR REPLACE VIEW ${myuniversity}_${mymodule}.group_budgets_view
```
[the remark this answers is not in the document]
Comment (craigmcnally) [reply src/main/resources/templates/db_scripts/group_budget_view.sql]: interesting.  OK let's just go with what you have here for now - hopefully this view is just a temporary solution anyway

File: src/main/resources/templates/db_scripts/schema.json (hunk 1 above)
Comment (craigmcnally) [reply src/main/resources/templates/db_scripts/schema.json]: I think this is all backwards compatible, so we should be ok w/o major version changes

Comment (pautau-s) [reply src/main/resources/templates/db_scripts/schema.json]: I meant a changes compared to [version 3.0.0](https://github.com/folio-org/mod-finance-storage/compare/v3.0.0...736421a0cd8353e289a69928a4ede7b3d13d0694).

**Pull request merged**